### ISSUE TITLE
Passing an invalid update Uint8array to Y.applyUpdate causes infinite loop in lib0 

### ISSUE THREAD

**Issue by jamesopti** [yjs/yjs#479]
**Describe the bug**
`Y.applyUpdate` can be passed an invalid Uint8array which causes the operation to lock the thread in an infinite loop.

The root cause is described in this lib0 issue: https://github.com/dmonad/lib0/issues/42

I understand the position that using lib0 directly can cause this (its an expert use library), but it feels brittle that `Y.applyUpdate` can easily trigger this as well.

At a minimum, it would be nice to have a utility to allow consumers to optionally validate their update before passing it in to a possibly unrecoverable operation.

**To Reproduce**

```ts
const JSBase64 = require("js-base64");
const Y = require("yjs");
const lib0 = require("lib0");

const sampleYdoc = new Y.Doc();
sampleYdoc.getArray("myarray").insert(0, ["Hello doc2, you got this?"]);
const sampleYdocState = Y.encodeStateAsUpdate(sampleYdoc); // is a Uint8Array
const sampleYdocString = JSBase64.fromUint8Array(sampleYdocState);

const updateGood = Buffer.from(sampleYdocString, "base64");
const updateBad = Buffer.from("garbagedoc", "base64");
 
Y.applyUpdate(new Y.Doc(), updateGood); // Works fine
Y.applyUpdate(new Y.Doc(), updateBad); // Hangs forever
```

**Expected behavior**

`Y.applyUpdate(new Y.Doc(), updateBad)` throws an error

**Environment Information**
 - YJS (latest)

**Comment by NilSet** [comment]
This also happens with other decode operations. Notably it happened to me when attempting to decode a state vector as if it were a snapshot.

IMO applyUpdate needs to be able to process bad data without a denial of service, as updates are definitely user controlled data you would be processing in a backend.

**Comment by dmonad** [comment]
This should be fixed when upgrading to `lib0@0.2.54`

**Comment by dmonad** [comment]
Great, happy to hear that!
